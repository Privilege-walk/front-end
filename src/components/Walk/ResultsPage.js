import React from "react";
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';

export default function ResultsPage(){
    return (
        <Container>
            <Typography variant="h3" component="div">User Results page</Typography>
        </Container>
    );
}