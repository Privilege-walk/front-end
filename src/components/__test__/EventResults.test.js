import React from "react";
import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import ResultsPage from '../Walk/ResultsPage';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Router from "react-router-dom";
const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

const mockedUsedNavigate = jest.fn();
const mockedUsedParams = jest.fn();
const mockFetchAnswerStats = jest.fn();
const mockFetchEventStats = jest.fn();
const mockUseParams = jest.fn();

jest.mock("react-router-dom", () => ({
    useParams: mockUseParams,
}));
jest.mock('../../Store/actions', () => ({
    fetchAnswerStats: () => mockFetchAnswerStats,
    fetchEventStats: () => mockFetchEventStats
}));
jest.mock("react-router-dom", () => ({
    Navigate: () => "Navigate",
    useNavigate: () => mockedUsedNavigate,
    useParams: () => mockedUsedParams
}))

test('renders without crashing',  () => {
    let utils;
    const store = mockStore({
        token: "token",
    });
    mockFetchAnswerStats.mockReturnValue({payload: {data:[{'question':1}]}});
    mockFetchEventStats.mockReturnValue({payload: {data:{
        'x_label_min':'x_label_min',
        'x_label_max':'x_label_max',
        'data': []
    }}});
    mockUseParams.mockReturnValue({ uniqueCode: '1234' });
    act (() => {
        utils = render( 
            <Provider store={store}>
                <ResultsPage />
            </Provider>
        );
    }); 
    const { asFragment } = utils;
    expect(asFragment()).toMatchSnapshot();
})