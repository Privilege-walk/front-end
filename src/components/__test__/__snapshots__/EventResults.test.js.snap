// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiGrid-root css-1b6z6ql-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-19vs8dh-MuiGrid-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Perspective Walk
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      />
    </div>
  </div>
</DocumentFragment>
`;
